pkg/app/server/grpcapi/web_api.go

```diff
@@ -501,8 +501,7 @@ func (a *WebAPI) AddApplication(ctx context.Context, req *webservice.AddApplicat
 		ProjectId:        claims.Role.ProjectId,
 		GitPath:          gitpath,
 		Kind:             req.Kind,
-		CloudProvider:    req.CloudProvider,
```

Comment:
@khanhtc1202 We have to keep this too.

Reply:
Not sure why do we need to keep this, since the web-ui will be updated when update the pipecd version, isn't it?

Reply:
Yes, but the old Pipeds only know the old CloudProvider field. So the newly added applications will not be handled correctly.

Reply:
I get your point 👍  In such case, this PR was a miss, since I remove the CloudProvider field from all application forms and add the new PlatformProvider field. I think the users should aware of what're they selecting on create new application. wdyt?

Reply:
If not, we have to keep both CloudProvider and PlatformProvider fields in application adding form? 🤔

Reply:
Actually, it was not deleted but just renamed in the eyes of end users, right?
So I feel that it could be fine. From the user's point of view, it's just a renaming.

Reply:
Sorry, I don't understand what do you mean by it could be fine 🙏 But lets me change this part like
```golang
		CloudProvider: req.PlatformProvider,
		PlatformProvider: req.PlatformProvider,
```
hence the application model's CloudProvider will be remained and keep as same as PlatformProvider value. I think it will resolve the fail case you mentioned 🤔

Reply:
addressed by https://github.com/pipe-cd/pipecd/pull/3821/commits/0ef103520eb13617d13cada263a4184d0818b067 🙏